src/charm.py

```diff
@@ -191,6 +192,16 @@ def __init__(self, *args):
         self.framework.observe(self.on.show_config_action, self._on_show_config_action)
         self.framework.observe(self.on.check_config_action, self._on_check_config)
 
+    @property
+    def self_scraping_job(self):
+        """The self-monitoring scrape job."""
+        port = urlparse(self._external_url).port or 80
```

Comment:
@dstathis you probably want to do something similar with Loki:
https://github.com/canonical/loki-k8s-operator/blob/c95a7ae42b1b4585aaf485caa2a3e0b512475a0d/src/charm.py#L102